cluster/node-pools/worker-splitaz/userdata.yaml

```diff
@@ -107,6 +107,9 @@ write_files:
           cacheAuthorizedTTL: "5m"
           cacheUnauthorizedTTL: "30s"
       protectKernelDefaults: true
+      # variables are replaced on instance start-up.
+      providerID: __PROVIDER_ID__
```

Comment:
According to the [docs](https://kubernetes.io/docs/tasks/administer-cluster/running-cloud-controller/#running-cloud-controller-manager) `kubelet` should also specify `--cloud-provider=external`. Maybe implicit when setting `providerID` but something to double check.

Reply:
This is also [done in here](https://github.bus.zalan.do/teapot/kubernetes-on-ubuntu/pull/332) it seems.